src/proxy.rs

```diff
@@ -165,28 +167,18 @@ impl ProxyHandlerWebSocket {
         // Stream frontend messages to backend.
         let stream_to_backend = async move {
             while let Some(Ok(msg_axm)) = frontend_stream.next().await {
-                // Ugh this is a lot of work that could be avoided if
-                // there were some way to get the wrapped message out
-                // directly, but there isn't...
-                let msg_tng = if msg_axm.is_binary() {
-                    MsgTng::Binary(msg_axm.as_bytes().to_owned())
-                } else if let Some(str) = msg_axm.to_str() {
-                    MsgTng::Text(str.to_owned())
-                } else if msg_axm.is_ping() {
-                    MsgTng::Ping(msg_axm.as_bytes().to_owned())
-                } else if msg_axm.is_pong() {
-                    MsgTng::Pong(msg_axm.as_bytes().to_owned())
-                } else if let Some((code, reason)) = msg_axm.close_frame() {
-                    MsgTng::Close(Some(CloseFrame {
-                        code: code.into(),
-                        reason: Cow::Owned(reason.to_owned()),
-                    }))
-                } else if msg_axm.is_close() {
-                    MsgTng::Close(None)
-                } else {
-                    tracing::error!(msg = ?msg_axm, "Received message is not binary, text, ping, pong, or close");
-                    return;
+                let msg_tng = match msg_axm {
+                    MsgAxm::Text(msg) => MsgTng::Text(msg.to_owned()),
```

Comment:
I don't think this `to_owned` is necessary anymore, is it?  It's already an owned `String` in the axum message, so it can just be moved, rather than copying its underlying `str`, same as for a `Binary` message's byte vec.

Reply:
Good catch! Fixed